### PR TITLE
Add option for access token in VertexAI

### PR DESCRIPTION
**Pull request by danielfrg**
Adds prop to directly pass an access token to the VertexAI LLMs.

This is useful in contexts where authentication is already handled in another service and an access token is returned. For example in Firebase we can easily get an access token for a particular user.

### CODE REVIEW

**Comment by jacoblee93** [comment]
Seems good to me! CC @afirstenberg

File: langchain/src/util/googlevertexai-webauth.ts [hunk 1]
```diff
@@ -35,7 +38,15 @@ export class WebGoogleAuth implements GoogleVertexAIAbstractedClient {
   }
 
   async request(opts: { url?: string; method?: string; data?: unknown }) {
-    const accessToken = await getAccessToken(this.options);
+    let { accessToken } = this.options;
+
+    if (accessToken === undefined) {
+      const opts = {
+        ...this.options,
+        credentials: this.options.credentials,
```
Comment (afirstenberg) [inline langchain/src/util/googlevertexai-webauth.ts:46]: Overall this looks good, but I think I'm missing why we're explicitly setting `credentials` here.

Comment (danielfrg) [reply langchain/src/util/googlevertexai-webauth.ts:46]: Still needed to get the project id to get the URL.

Comment (danielfrg) [reply langchain/src/util/googlevertexai-webauth.ts:46]: Oh sorry, missunderstood the question. I dont think we need that. I kept it was already there. I can remove if you want.

Comment (afirstenberg) [reply langchain/src/util/googlevertexai-webauth.ts:46]: Unless @jacoblee93 sees a reason, I'd say to simplify it back to using this.options.

Comment (jacoblee93) [reply langchain/src/util/googlevertexai-webauth.ts]: Done

**Comment by jacoblee93** [comment]
Thank you!

**Pull request merged**